src/main/java/net/neoforged/neoforge/event/entity/EntityTeleportEvent.java

```diff
@@ -75,33 +84,47 @@ public void setTargetZ(double targetZ) {
     }
 
     public Level getTargetLevel() {
-        return target.level;
+        return target.getLevel();
     }
 
     @ApiStatus.Internal
     // Casting helper. Backup case shouldn't ever happen, see protection in setTargetLevel().
-    public ServerLevel getTargetServerlevelOr(ServerLevel backup) {
-        return target.level instanceof ServerLevel serverlevel ? serverlevel : backup;
+    public ServerLevel getTargetServerLevelOr(ServerLevel backup) {
+        return target.getLevel(backup) instanceof ServerLevel serverlevel ? serverlevel : backup;
     }
 
+    // TODO: Update Javadoc when the deprecated constructors get removed 
     /**
      * Checks if the event sub-type support changing the target level.
      * <p>
-     * Note that a positive result is not a guarantee that the change will be honored by a modded mode of teleportation. We do not recommend changing the target level when original source and target levels are the same.
+     * 
+     * @apiNote At the moment, a positive result is not a guarantee that the change will be honored by a modded mode of teleportation.
+     *          Only changing the target level when the original target level is different from the source level can be considered safe in 1.21.1.
+     * @implSpec This needs to be overwritten by subclasses depending on the capabilities of the code firing the event.
```

Comment:
Technically, the term is "method overriding", so this should be "to be overridden". 😄